test/common/http/path_utility_test.cc

```diff
@@ -57,7 +75,9 @@ TEST_F(PathUtilityTest, NormalizeValidPaths) {
       {"/a/..\\c", "/c"},           // "..\\" canonicalization
       {"/%c0%af", "/%c0%af"},       // 2 bytes unicode reserved characters
       {"/%5c%25", "/%5c%25"},       // reserved characters
-      {"/a/b/%2E%2E/c", "/a/c"}     // %2E escape
+      {"/a/b/%2E%2E/c", "/a/c"},    // %2E escape
+      {"/a/b/%2E./c", "/a/c"},      {"/a/b/%2E/c", "/a/b/c"}, {"..", "/"},
```

Comment:
nit: could you add a test for `.%2E` sequence too?

Reply:
Should paths with other "reserved" characters be tested too (technically `.` isn't reserved, but we treat it like one)? For instance, a %-encoded `?` could trip query detection; %-encoded `/` is allowed within path and isn't supposed to be treated like a path separator...

Reply:
Thanks!